packages/components/src/range-control/input-range.js

```diff
@@ -23,23 +34,30 @@ function InputRange(
 		label,
 		onHideTooltip = noop,
 		onMouseLeave = noop,
-		step = 1,
 		onBlur = noop,
 		onChange = noop,
 		onFocus = noop,
 		onMouseMove = noop,
+		onShiftStep,
 		onShowTooltip = noop,
-		shiftStep = 10,
+		shiftStep,
 		value,
 		...props
 	},
 	ref
 ) {
-	const jumpStep = useJumpStep( {
-		step,
-		shiftStep,
-		isShiftStepEnabled,
-	} );
+	const onKeyDown = ( event ) => {
+		const { keyCode, shiftKey } = event;
+		props.onKeyDown?.( event );
+
+		if ( isShiftStepEnabled && shiftKey && keyCode in operationList ) {
+			event.preventDefault();
+			const { min, max, step } = props;
+			const modifiedStep = shiftStep * step;
+			const nextValue = operationList[ keyCode ]( value, modifiedStep );
+			onShiftStep( roundClamp( nextValue, min, max, modifiedStep ) );
```

Comment:
Is the clamping here necessary?

If `onShiftStep` is assigned to the `handleOnChange` function, the clamping should already happen in there when [calling `setValue`](https://github.com/WordPress/gutenberg/blob/6b780e7e5cd025951a9145dea2a1a0898aec6e78/packages/components/src/range-control/index.js#L126) (which comes from [the `useControlledRangeValue` hook](https://github.com/WordPress/gutenberg/blob/5cca2e1a8021679aaf2fc713c1a2c1cce168c374/packages/components/src/range-control/utils.js#L52)).

Reply:
Good eye! `roundClamp` was added only for its rounding. If it's preferred I could look at changing this to `round` from lodash or maybe inlining the expression we need. Or perhaps this makes a case for having the `useControlledRangeValue` hook made `step`-aware so it could handle the rounding. Though that might be better explored as part of a PR for #34817.